crates/ty_python_semantic/resources/mdtest/narrow/match.md

```diff
@@ -105,15 +134,15 @@ x = get_object()
 reveal_type(x)  # revealed: object
 
 match x:
-    case "foo" if reveal_type(x):  # revealed: Literal["foo"]
+    case "foo" if reveal_type(x):  # revealed: object
```

Comment:
I think we should update these tests here to see that `x` in the guard expression *can* be a narrowed type.